src/cdk-experimental/menu/menu-item-trigger.ts

```diff
@@ -24,29 +42,121 @@ import {CdkMenuItem} from './menu-item';
+    protected readonly _viewContainerReference: ViewContainerRef,
+    private readonly _overlay: Overlay,
+    private readonly _directionality: Directionality,
+    @Inject(CDK_MENU) private readonly _parentMenu: Menu
   ) {}
 
-  ngAfterContentInit() {
-    this._setHasSubmenu();
-  }
-
-  /** Set the hasSubmenu property on the menuitem  */
-  private _setHasSubmenu() {
-    if (this._menuItemInstance) {
-      this._menuItemInstance.hasSubmenu = this._hasSubmenu();
+  /** Open/close the attached submenu if the trigger has been configured with one */
+  toggle() {
+    if (this.hasSubmenu()) {
+      this.isSubmenuOpen() ? this._closeSubmenu() : this._openSubmenu();
     }
   }
+
   /** Return true if the trigger has an attached menu */
-  private _hasSubmenu() {
+  hasSubmenu() {
     return !!this._menuPanel;
   }
+
+  /** Whether the submenu this button is a trigger for is open */
+  isSubmenuOpen() {
+    return this._overlayReference ? this._overlayReference.hasAttached() : false;
+  }
+
+  /** Open the attached submenu */
+  private _openSubmenu() {
+    this.opened.next();
+
+    this._overlayReference = this._overlay.create(this._getOverlayConfig());
+    this._overlayReference.attach(this._getPortal());
+  }
+
+  /** Close the opened submenu */
+  private _closeSubmenu() {
+    if (this.isSubmenuOpen()) {
+      this.closed.next();
+
+      this._overlayReference!.detach();
+    }
+  }
+
+  /** Return the configuration object used to create the overlay */
+  private _getOverlayConfig() {
+    return new OverlayConfig({
+      positionStrategy: this._getOverlayPositionStrategy(),
+      scrollStrategy: this._overlay.scrollStrategies.block(),
+      direction: this._directionality,
+    });
+  }
+
+  /** Build the position strategy for the overlay which specifies where to place the submenu */
+  private _getOverlayPositionStrategy(): FlexibleConnectedPositionStrategy {
+    return this._overlay
+      .position()
+      .flexibleConnectedTo(this._elementReference)
+      .withPositions(this._getOverlayPositions());
+  }
+
+  /** Determine and return where to position the submenu relative to the menu item */
+  private _getOverlayPositions(): ConnectedPosition[] {
+    return this._parentMenu.orientation === 'horizontal'
+      ? [
+          {originX: 'start', originY: 'bottom', overlayX: 'start', overlayY: 'top'},
+          {originX: 'start', originY: 'top', overlayX: 'start', overlayY: 'bottom'},
+          {originX: 'end', originY: 'bottom', overlayX: 'end', overlayY: 'top'},
+          {originX: 'end', originY: 'top', overlayX: 'end', overlayY: 'bottom'},
+        ]
+      : [
+          {originX: 'end', originY: 'top', overlayX: 'start', overlayY: 'top'},
+          {originX: 'end', originY: 'bottom', overlayX: 'start', overlayY: 'bottom'},
+          {originX: 'start', originY: 'top', overlayX: 'end', overlayY: 'top'},
+          {originX: 'start', originY: 'bottom', overlayX: 'end', overlayY: 'bottom'},
```

Comment:
It would be good to add a `TODO` here to one common "source of truth" for the conventional menu/dropdown positioning strategy, potentially in `cdk/overlay`